Sprint_6

ArmMart0896 opened the pull request.

Commit:
Выполненное задание 6-го спринта

Commit:
add jacoco report

mkouznetsov-git commented on src/test/java/com/example/LionManeTest.java:
```diff
@@ -0,0 +1,49 @@
+        return new Object[][]{
+                {"Самец", true},
+                {"Самка", false},
+                {"Самей", false},
```
Некорректно ожидать false или true для неадекватных значений пола. В этих случаях должен быть Exception. И такие случаи следует вынести в отдельный тестовый класс.

mkouznetsov-git commented on src/test/java/com/example/LionManeTest.java:
```diff
@@ -0,0 +1,49 @@
+        };
+    }
+    @Test
+    public void doesLionHaveManeTest() {
```
В таком тесте, если вдруг случится Exception для корректного значения пола, то мы не отловим это, т.к. тест будет успешно проходить. А он должен будет упасть.

mkouznetsov-git commented on src/test/java/com/example/LionManeTest.java:
```diff
@@ -0,0 +1,49 @@
+            Assert.assertEquals(isMane, lion.doesHaveMane());
+        }
+        catch (Exception exception) {
+            System.out.println("Указан недопустимый пол животного");
```
Вывод в консоль не следует использовать в авто-тестах, его никто не будет читать. Для отладки лично для себя можно это делать, но в финальной версии кода такое не оставлять не надо.

Commit:
Выполненное задание 6-го спринта с доработками.
1) Для неадекватных значений создал еще один тестовый класс. Так же определил класс LionException внутри класса Lion;
2) Попробовал исправить код в соответствии со вторым замечанием. Добавил к аннотации @Test параметр expected;
3) В новом тестовом классе вместо System.out.println воспользовался Assert.fail, вызывающее сообщение

mkouznetsov-git commented on src/test/java/com/example/LionManeTest.java:
```diff
@@ -0,0 +1,44 @@
+
+        };
+    }
+    @Test (expected = Lion.LionException.class)
```
Почему мы здесь должны ждать exception?

ArmMart0896 replied:
Запутался с этими исключениями, подумал, что падение теста будет ожидаемым в этом задании.
Убрал это из аннотации. без этого тесты не падают

mkouznetsov-git commented on src/test/java/com/example/LionIncorrectSexTest.java:
```diff
@@ -0,0 +1,48 @@
+        };
+    }
+    @Test(expected = Lion.LionException.class)
+    public void incorrectSexThrowsExceptionTest() throws Exception {
```
Тест падает ещё до проверок:
java.lang.IllegalArgumentException: wrong number of arguments

ArmMart0896 replied:
Извиняюсь. Перемудрил видимо

Commit:
Выполненное задание 6-го спринта с новыми доработками.
